src/main/webapp/app/course/manage/detail/course-detail-bar-chart.component.ts

```diff
@@ -0,0 +1,164 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { ChartDataSets, ChartType } from 'chart.js';
+import { Label } from 'ng2-charts';
+import { TranslateService } from '@ngx-translate/core';
+import * as moment from 'moment';
+import { Graphs } from 'app/entities/statistics.model';
+import { CourseManagementService } from '../course-management.service';
+
+@Component({
+    selector: 'jhi-course-detail-bar-chart',
+    templateUrl: './course-detail-bar-chart.component.html',
+    styleUrls: ['./course-detail-bar-chart.component.scss'],
+})
+export class CourseDetailBarChartComponent implements OnChanges {
+    @Input()
+    courseId: number;
+    @Input()
+    numberOfStudentsInCourse: number;
+    @Input()
+    initialStats: number[];
+    initialStatsReceived = false;
+
+    graphType: Graphs = Graphs.ACTIVE_STUDENTS;
+
+    LEFT = false;
+    RIGHT = true;
+    Graphs = Graphs;
+
+    // Chart
+    chartTime: any;
+    // Histogram related properties
+    barChartOptions: any = {};
+    barChartType: ChartType = 'line';
+    amountOfStudents: string;
+    barChartLegend = false;
+    // Data
+    barChartLabels: Label[] = [];
+    chartData: ChartDataSets[] = [];
+    data: number[] = [];
+
+    // Left arrow -> decrease, right arrow -> increase
+    private currentPeriod = 0;
+
+    constructor(private service: CourseManagementService, private translateService: TranslateService) {}
+
+    ngOnChanges() {
```

Comment:
ngOnChanges() and reloadChart() are basically the same code, is it necessary to keep them apart?